lua/neotest/consumers/state/internal.lua

```diff
@@ -0,0 +1,208 @@
+local au_result = "NeotestResult"
+local au_started = "NeotestStarted"
+
+---@class neotest.StateConsumer
+---@field private _result table<integer, table<string, neotest.Tree> >
+---@field private _status table
+---@field private _cache table
+---@field private _running table<string, table<string, string>>
+---@field private _adapters table<integer, string>
+---@field private _client neotest.Client
+local state = {
+  _result = {},
+  _status = {},
+  _cache = {},
+  _running = {},
+  _adapters = {},
+}
+
+---Escape characters that have a special meaning in patterns
+---@param text string
+---@return string
+local function escape(text)
+  text = text:gsub("%-", "%%-")
+  text = text:gsub("%.", "%%.")
+  return text
+end
+
+---Perform a double fuzzy match on two strings
+---@param a string
+---@param b string
+---@return boolean
+local function fuzzy_match(a, b)
+  return string.match(a, escape(b)) or string.match(b, escape(a))
+end
+
+---Update the internal list of currently running tests
+---@param adapter_id string
+---@param position_ids string
+function state:_update_running(adapter_id, position_ids)
+  self._running[position_ids] = {
+    adapter = string.match(adapter_id, "(.-):"),
+    path = position_ids,
+  }
+  -- Trigger Event
+  vim.api.nvim_exec_autocmds(
+    "User",
+    { pattern = au_started, data = { adapter_id = adapter_id, running = position_ids } }
+  )
+end
+
+---Process results and count pass/fail rates, save to cache
+---Passed, Failed, Skipped
+---@param tree neotest.Tree|nil
+function state:_process_result(tree, results)
+  if not tree then
+    return
+  end
+  for _, pos in tree:iter() do
+    if pos.type == "test" then
+      if results[pos.id].status then
+        self:_set_status(pos.path, pos.name, results[pos.id].status)
+      end
+    end
+  end
+  self:_update_cache()
+end
+
+---Update status for specific test
+---@param path string
+---@param status string
+function state:_set_status(path, name, status)
+  if self._status[path] == nil then
+    self._status[path] = {}
+  end
+  self._status[path][name] = status
+end
+
+---Update the cache status count
+function state:_update_cache()
+  local total = { passed = 0, failed = 0, skipped = 0, unknown = 0 }
```

Comment:
`unknown` is not a valid status, tests are treated as unknown if they have no result recorded and are not running so this will always be 0

Reply:
Currently yes. I had put this in as future work. I thought it could be nice to also discover unrun tests in the future, so you can display them as well. As I personally didn't care too much for it yet I haven't spend the time to implement it